service/src/main/python/jcarbon/tensorflow/callbacks.py

```diff
@@ -185,3 +188,100 @@ def on_epoch_end(self, epoch, logs=None):
         self.client.stop(self.pid)
         self.client.dump(
             self.pid, f'{self.output_path}/report-{epoch}.pb', self.signals)
+
+class JCarbonNvmlCallback(JCarbonCallback):
+    def __init__(
+            self,
+            addr='localhost:8980',
+            period_ms=DEFAULT_PERIOD_MS,
+            signals=DEFAULT_SIGNALS,
+            chunking_period_sec=DEFAULT_PERIOD_SECS):
+        super().__init__(addr, period_ms, signals)
+        self.reports = {}
+        self.sampler = NvmlSampler()
+    
+    def on_epoch_begin(self, epoch, logs = None):
+        self.last_report = None
+        self.sampler.sample()
+    
+    def on_epoch_end(self, epoch, logs = None):
+        self.sampler.sample()
+        if self.last_report is None:
+            self.last_report = [create_report(self.sampler.samples)]
+            print('in none)')
+            print(self.last_report)
+        else:
+            self.last_report.append(create_report(self.sampler.samples))
+            print(self.last_report)
+
+        self.reports[epoch] = pd.concat(list(map(
+            to_dataframe,
+            self.last_report
+        ))).to_frame().assign(
+            epoch=epoch).set_index('epoch', append=True)
+
+class JCarbonChunkingCallback3(JCarbonCallback):
```

Comment:
This should be a subclass of `JCarbonChunkingCallback`. You are rewriting an enormous amount of code. You should be able to call the superclass method using `super()`.

Reply:
got it, could you help check if my callback looks correct? it is getting the output we wanted though